core/include/thread_flags.h

```diff
@@ -61,13 +65,40 @@
  * @name reserved thread flags
  * @{
  */
+
+#if defined(MODULE_CORE_THREAD_FLAGS) || defined(DOXYGEN)
```

Comment:
Should not be necessary to check for MODULE_CORE_THREAD_FLAGS since we are in the thread_flags.h header.
If you want to add additional checks you could do 
```c
#ifndev MODULE_CORE_THREAD_FLAGS
#error Missing USEMODULE += core_thread_flags
#endif
```

Reply:
Good idea!

Reply:
done